logstash-core/src/main/java/org/logstash/Valuefier.java

```diff
@@ -120,8 +118,7 @@ private static Object fallbackConvert(final Object o, final Class<?> cls) {
         converters.put(
             BigDecimal.class, value -> new RubyBigDecimal(RubyUtil.RUBY, (BigDecimal) value)
         );
-        converters.put(Long.class, LONG_CONVERTER);
-        converters.put(Integer.class, LONG_CONVERTER);
+        converters.put(Number.class, input -> RubyUtil.RUBY.newFixnum(((Number) input).longValue()));
```

Comment:
@msvticket 

This is not a good change. The key for the converters map needs to a concrete class. The `o.getClass()` will always be the concrete class so the lookup will not find this Number.class key, forcing the fallback execution.
I think adding `converters.put(Short.class, LONG_CONVERTER);` is all that is needed (and the Date one below) then we have all the Number subclasses covered.